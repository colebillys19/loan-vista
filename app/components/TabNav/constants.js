export const VALID_ROUTES = ['/calls', '/documents', '/payments', '/escrow'];
