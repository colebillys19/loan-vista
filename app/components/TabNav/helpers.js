import { VALID_ROUTES } from './constants';

export const checkRoute = (pathname) => {
  if (pathname === '/' || VALID_ROUTES.indexOf(pathname) === -1) {
    return false;
  }

  return VALID_ROUTES.indexOf(pathname);
};
