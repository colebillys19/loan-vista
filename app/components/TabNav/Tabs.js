import React from 'react';
import T from 'prop-types';

import { StyledTab, StyledTabs } from './styledComponents';

const Tabs = ({ handleChange, value }) => (
  <StyledTabs value={value} onChange={handleChange}>
    <StyledTab aria-controls="calls-view" disableRipple id="calls-tab" label="Calls" />
    <StyledTab aria-controls="emails-view" disableRipple id="emails-tab" label="Emails" />
    <StyledTab aria-controls="payments-view" disableRipple id="payments-tab" label="Payments" />
    <StyledTab aria-controls="misc-view" disableRipple id="misc-tab" label="Misc" />
  </StyledTabs>
);

Tabs.propTypes = {
  handleChange: T.func.isRequired,
  value: T.oneOfType([T.number, T.bool]).isRequired,
};

export default Tabs;
