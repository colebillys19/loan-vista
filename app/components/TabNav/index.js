/**
 * TabNav
 * @description ...
 */

import React, { useState } from 'react';

import Tabs from './Tabs';
import {
  ContentWrapper,
  HomePageButtonWrapper,
  StyledButton,
  TabNavWrapper,
} from './styledComponents';

const TabNav = () => {
  const [value, setValue] = useState(0);

  const handleChange = (e, newValue) => {
    setValue(newValue);
  };

  return (
    <TabNavWrapper>
      <ContentWrapper>
        <HomePageButtonWrapper>
          <StyledButton disableRipple>Loan Profile</StyledButton>
        </HomePageButtonWrapper>
        <Tabs value={value} handleChange={handleChange} />
      </ContentWrapper>
    </TabNavWrapper>
  );
};

export default TabNav;
