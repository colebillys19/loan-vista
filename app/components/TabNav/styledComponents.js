import styled from 'styled-components';
import Button from '@material-ui/core/Button';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';

import { mainBlue } from 'styleConstants';

export const ContentWrapper = styled.div`
  height: 6rem;
  margin: 0 auto;
  width: 120rem;
`;

export const HomeViewButtonWrapper = styled.div`
  align-items: center;
  display: flex;
  position: absolute;
  width: 36rem;
`;

export const StyledButton = styled(Button)`
  display: block;
  font-size: 2.2rem;
  height: 6rem;
  margin: 0 auto;
  padding: 0;
  text-transform: none;
  & :hover {
    text-decoration: underline;
    background-color: transparent;
  }
`;

export const StyledTab = styled(Tab)`
  flex-grow: 1;
  font-size: 1.4rem;
  height: 6rem;
  max-width: 22rem;
  text-transform: none;
`;

export const StyledTabs = styled(Tabs)`
  display: inline-block;
  left: 36.3rem;
  position: relative;
  width: 83.7rem;
  & .MuiTabs-indicator {
    background-color: ${mainBlue};
    height: 0.5rem;
  }
`;

export const TabNavWrapper = styled.div`
  margin-bottom: 0.3rem;
`;
