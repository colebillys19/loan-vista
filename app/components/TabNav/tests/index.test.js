import React from 'react';
import { render } from 'react-testing-library';

import TabNav from '../index';

const mockProps = { dispatchNavigation: jest.fn(), pathname: '/calls' };

const Component = <TabNav {...mockProps} />;

describe('<TabNav />', () => {
  it('Expect not to log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(Component);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });
});
