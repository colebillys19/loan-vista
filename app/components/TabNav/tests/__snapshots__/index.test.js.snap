// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabNav /> Should render and match the snapshot 1`] = `
<div
  class="styledComponents__TabNavWrapper-ift0se-3 hseaqD"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text styledComponents__StyledButton-ift0se-0 labhVu"
    id="dashboard-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Dashboard
    </span>
  </button>
  <div
    class="MuiTabs-root styledComponents__StyledTabs-ift0se-2 dNoHxc"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed"
      style="overflow: hidden;"
    >
      <div
        class="MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-controls="calls-view"
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit styledComponents__StyledTab-ift0se-1 eCObik Mui-selected"
          id="calls-tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            Calls
          </span>
        </button>
        <button
          aria-controls="documents-view"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit styledComponents__StyledTab-ift0se-1 eCObik"
          id="documents-tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            Documents
          </span>
        </button>
        <button
          aria-controls="payments-view"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit styledComponents__StyledTab-ift0se-1 eCObik"
          id="payments-tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            Payments
          </span>
        </button>
        <button
          aria-controls="escrow-view"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit styledComponents__StyledTab-ift0se-1 eCObik"
          id="escrow-tab"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            Escrow
          </span>
        </button>
      </div>
      <span
        class="PrivateTabIndicator-root-5 PrivateTabIndicator-colorSecondary-7 MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</div>
`;
